src/main/java/com/google/sps/data/DataStoreManager.java

```diff
@@ -0,0 +1,242 @@
+    game.setNumStarVotes(numStarVotes).setTotalStars(totalStars).setNumDifficultyVotes(numDifficultyVotes).setTotalDifficulty(totalDifficulty);       
+    return game.build();
+  }
+
+  /**
+    * Stores the static data of a single stage in datastore as an entity
+    * @param game an Stage variable representing a single instance of a stage.
+  */
+  public void storeStage(Stage stage){
+    Entity stageEntity = new Entity("Stage", stage.getStageID());
+    stageEntity.setProperty("stageNumber", stage.getStageNumber());
+    stageEntity.setProperty("key", stage.getKey());
+    stageEntity.setProperty("startingHint", stage.getStartingHint());
+    stageEntity.setProperty("latitude", stage.getStartingLocation().getLatitude());
+    stageEntity.setProperty("longitude", stage.getStartingLocation().getLongitude());
+    for (Hint hint : stage.getHints())
+    {
+      storeHints(hint);
+    }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(stageEntity);
+  }
+  
+  /**
+    * Updates an entity of a single stage data in datastore
+    * @param stage a Stage variable representing a single instance of a stage.
+  */
+  public void updateStage(Stage stage){
+    Key stageEntityKey = KeyFactory.createKey("Stage", stage.getStageID());
+    try {
+      Entity stageEntity = datastore.get(stageEntityKey);
+    }
+    catch(Exception e) {
+      return;
+    }
+      
+    stageEntity.setProperty("stageNumber", stage.getStageNumber());
+    stageEntity.setProperty("key", stage.getKey());
+    stageEntity.setProperty("startingHint", stage.getStartingHint());
+    stageEntity.setProperty("latitude", stage.getStartingLocation().getLatitude());
+    stageEntity.setProperty("longitude", stage.getStartingLocation().getLongitude());
+    for (Hint hint : stage.getHints())
+    {
+      updateHints(hint);
+    }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(stageEntity);
+  }
+  
+  /**
+    * Retrieves a single stage entity from the datastore.
+    * @param stageID the unique code used to identify this specific stage.
+    * @return a Stage object with the properties specified in the Builder.
+  */
+  public Game retrieveStage(String stageID){
+    Key stageEntityKey = KeyFactory.createKey("Stage", stageID);
+    Entity stageEntity = datastore.get(stageEntityKey);
+
+    int stageNumber = (int) stageEntity.getProperty("stageNumber");
+    String key = (String) stageEntity.getProperty("key");
+    String startingHint = (String) stageEntity.getProperty("startingHint");
+    double latitude = (double) stageEntity.getProperty("latitude");
+    double longitude = (double) stageEntity.getProperty("longitude");
+    Coordinates startingLocation;
+    startingLocation.Coordinates(latitude, longitude);
+    ArrayList<Hint> hints = stageEntity.getProperty("hints");
```

Comment:
This won't work because stageEntity does not have a property called "hints". You would need to store for each stage, a list of hintID of its hints. Then retrieve that list, iterate through it, and search for each hint within datastore by inputting its id.